src/Features/CSharp/Portable/UseLocalFunction/CSharpUseLocalFunctionCodeFixProvider.cs

```diff
@@ -46,111 +46,128 @@ public override Task RegisterCodeFixesAsync(CodeFixContext context)
         {
             var semanticModel = await document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
 
-            var localDeclarationToLambda = new Dictionary<LocalDeclarationStatementSyntax, LambdaExpressionSyntax>();
+            var nodesFromDiagnostics = new List<(
+                LocalDeclarationStatementSyntax declaration,
+                AnonymousFunctionExpressionSyntax function,
+                List<InvocationExpressionSyntax> invocations)>();
```

Comment:
that said, I'm still not convinced that a tuple with 3 elements is the best thing to do here, but
1. i don't think this should be a named type, because
    1. i can't even come up with a name for it
    2. it's only used locally in this method
2. as i said, what other people might have thought about doing here is a dictionary or after this fix, 2 dictionaries, one from declaration to anonymous function, one from declaration to invocation. i think that's not the right solution because again, there's no actual lookup needed when you're just iterating through it. this is not what hash tables are for.